src/transforms.jl

```diff
@@ -254,7 +254,7 @@ function transform_yield(expr)
   _is_yield(expr) || return expr
   ret = length(expr.args) > 2 ? expr.args[3:end] : [nothing]
   quote
-    $(ret...)
+    Base.inferencebarrier($(ret...))
```

Comment:
could you comment why the inference barrier is needed only for this transformation